kamon-monix/src/test/scala/kamon/instrumentation/MonixInstrumentationSpec.scala

```diff
@@ -0,0 +1,93 @@
+package kamon.instrumentation.futures.cats
+
+import java.util.concurrent.Executors
+
+import kamon.Kamon
+import kamon.tag.Lookups.plain
+import kamon.context.Context
+import org.scalatest.{Matchers, OptionValues, WordSpec}
+import org.scalatest.concurrent.{Eventually, PatienceConfiguration, ScalaFutures}
+
+import scala.concurrent.ExecutionContext.global
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import monix.eval.Task
+import monix.execution.Scheduler
+
+class MonixInstrumentationSpec extends WordSpec with ScalaFutures with Matchers with PatienceConfiguration
+    with OptionValues with Eventually {
+
+  // NOTE: We have this test just to ensure that the Context propagation is working, but starting with Kamon 2.0 there
+  //       is no need to have explicit Runnable/Callable instrumentation because the instrumentation brought by the
+  //       kamon-executors module should take care of all non-JDK Runnable/Callable implementations.
+
+  "an Monix Task created when instrumentation is active" should {
+    "capture the active span available when created" which {
+      "must be available across asynchronous boundaries" in {
+        implicit val ctxShift: Scheduler = Scheduler(global)
+
+        val anotherExecutionContext: ExecutionContext =
+          ExecutionContext.fromExecutorService(Executors.newCachedThreadPool())
+        val context = Context.of("key", "value")
+        val contextTagAfterTransformations =
+          for {
+            scope <- Task {
+              Kamon.storeContext(context)
+            }
+            len <- Task("Hello Kamon!").map(_.length)
+            _ <- Task(len.toString)
+            _ <- Task.shift(global)
+            _ <- Task.shift
+            _ <- Task.shift(anotherExecutionContext)
+          } yield {
+            val tagValue = Kamon.currentContext().getTag(plain("key"))
+            scope.close()
+            tagValue
+          }
+
+        val contextTagFuture = contextTagAfterTransformations.runToFuture
+
+
+        eventually(timeout(10 seconds)) {
+          contextTagFuture.value.get.get shouldBe "value"
+        }
+      }
+
+      "must be available in parallel settings" in {
+        implicit val ctxShift: Scheduler = monix.execution.Scheduler.Implicits.traced
```

Comment:
Hey @dispalt, is this what you were referring to [on Gitter](https://gitter.im/kamon-io/Kamon?at=5e430de8f3718e705df77c6e)?

Reply:
yes, I don't fully grok why it works with traced implementation context, or how that effects monix, but I don't feel like it should be necessary to use that.  The first commit shows the test failing, this one shows it working with the traced context.